Sources/NIO/ChannelPipeline.swift

```diff
@@ -1380,7 +1380,6 @@ public final class ChannelHandlerContext: ChannelInvoker {
 
     fileprivate func invokeRegister(promise: EventLoopPromise<Void>?) {
         self.eventLoop.assertInEventLoop()
-        assert(promise.map { !$0.futureResult.isFulfilled } ?? true, "Promise \(promise!) already fulfilled")
```

Comment:
these totally didn't make any sense anyway because in all other places, we allow pre-fulfilled futures. This could also cause racy crashes.